app/controllers/application_controller.rb

```diff
@@ -1,2 +1,10 @@
 class ApplicationController < ActionController::Base
+  before_action :set_locale
+  def set_locale
+    locale = params[:locale].to_s.strip.to_sym
+    I18n.locale = I18n.available_locales.include?(locale) ?
+        locale :
+        I18n.default_locale
```

Comment:
Can we write this in a single line like this?
`I18n.locale = I18n.available_locales.include?(locale) ? locale : I18n.default_locale`